recipes/libboost-mpi/meta.yaml

```diff
@@ -0,0 +1,49 @@
+{% set version = "1.78.0" %}
+{% set filename = "boost_%s.tar.bz2" % version.replace(".", "_") %}
+
+package:
+  name: libboost-mpi
+  version: {{ version }}
+
+source:
+  url: https://boostorg.jfrog.io/artifactory/main/release/{{ version }}/source/{{ filename }}
+  sha256: 8681f175d4bdb26c52222665793eef08490d7758529330f98d3b29dd0735bccc
+  patches:
+    - 5eff1ecc8413b0dc93a1ab047d7fed751e6cb40e.patch  # [not win]
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+
+  host:
+    - {{ mpi }}
+
+  run:
+    - boost-cpp {{ version }}.*
```

Comment:
You need this in `host` too

Reply:
however, I should probably update to 1.79...

Reply:
To avoid the following
```
022-05-02T09:32:49.4804653Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-05-02T09:32:49.4807397Z   packages: conda-forge/linux-64::boost-cpp-1.78.0-h75c5d50_1, local/linux-64::libboost-mpi-1.78.0-h9b6f76d_0
2022-05-02T09:32:49.4808785Z   path: 'lib/libboost_serialization.so'
2022-05-02T09:32:49.4810037Z 
2022-05-02T09:32:49.4810680Z 
2022-05-02T09:32:49.4811320Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-05-02T09:32:49.4812413Z   packages: conda-forge/linux-64::boost-cpp-1.78.0-h75c5d50_1, local/linux-64::libboost-mpi-1.78.0-h9b6f76d_0
2022-05-02T09:32:49.4813428Z   path: 'lib/libboost_serialization.so.1.78.0'
2022-05-02T09:32:49.4814114Z 
2022-05-02T09:32:49.4814481Z 
2022-05-02T09:32:49.4815096Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-05-02T09:32:49.4821002Z   packages: conda-forge/linux-64::boost-cpp-1.78.0-h75c5d50_1, local/linux-64::libboost-mpi-1.78.0-h9b6f76d_0
2022-05-02T09:32:49.4822577Z   path: 'lib/libboost_wserialization.so'
2022-05-02T09:32:49.4823433Z 
2022-05-02T09:32:49.4823724Z 
2022-05-02T09:32:49.4824256Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-05-02T09:32:49.4825384Z   packages: conda-forge/linux-64::boost-cpp-1.78.0-h75c5d50_1, local/linux-64::libboost-mpi-1.78.0-h9b6f76d_0
2022-05-02T09:32:49.4826995Z   path: 'lib/libboost_wserialization.so.1.78.0'
```

Reply:
@isuruf Thanks for the head-up on the cloberring warnings in the test. I do not intend to install Boost.Serialization so cloberring should not happen. Since `install-dependencies=off`,  serialization _should not_ be installed in this package. Looks like either I did not understand the `install-dependencies` option or b2 is broken.

Reply:
After checking with the Boost team, I need to remove Boost.Seralization manually as passing `install-dependencies=off` on the command line is not going to work.

Reply:
It's better to have `boost-cpp` in `host` so that clobbering doesn't happen accidentally in the future when dependencies change.

Reply:
Ok reverting my latest changes and adding to host. I dont understand why having `boost-cpp` in `host` prevents cloberring though... I probably have a wrong definition of clobbering, overwriting?

Reply:
conda-build packages only files that are new. Since `libboost_serialization*` are already in host when `boost-cpp` is in host, they are not considered new and therefore are not packaged in the `libboost-mpi` conda package.

Reply:
Thanks for taking the time to explain, the cloberring is indeed gone.